File: examples/java/kora-java-scheduling-jdk/src/main/java/io/koraframework/example/scheduling/jdk/ConfigScheduler.java
package io.koraframework.example.scheduling.jdk;

import io.koraframework.common.annotation.Component;
import io.koraframework.scheduling.jdk.annotation.ScheduleAtFixedRate;

@Component
public final class ConfigScheduler {

    private int state = 0;

    @ScheduleAtFixedRate(config = "scheduling.jobs.fix-rate")
    void schedule() {
        state++;
    }

    public int getState() {
        return state;
    }
}

File: examples/kotlin/kora-kotlin-scheduling-jdk/src/main/kotlin/io/koraframework/kotlin/example/scheduling/jdk/ConfigScheduler.kt
package io.koraframework.kotlin.example.scheduling.jdk

import io.koraframework.common.annotation.Component
import io.koraframework.scheduling.jdk.annotation.ScheduleAtFixedRate
import io.koraframework.scheduling.jdk.annotation.ScheduleOnce
import io.koraframework.scheduling.jdk.annotation.ScheduleWithFixedDelay
import java.time.temporal.ChronoUnit

@Component
class ConfigScheduler {
    var state = 0
        private set

    @ScheduleAtFixedRate(config = "scheduling.jobs.fix-rate")
    fun schedule() {
        state++
    }
}

